### DISCUSSION TITLE
PythonAnywhere implementation with SQL-Alchemy

### DISCUSSION THREAD

**Discussion by ziscoplankton** [sqlalchemy/sqlalchemy#9088] [Usage Questions] [answered]
Hi everyone,

I have deployed a Flask app with SQL-Alchemy and unfortunately the WSGI says that it is not able to open the database file.

I have created a stackoverflow thread which you can find below. It include some details from the log files and when I try to run the wsgi.py from the var/www/path

[WSGI ERROR DETAILS](https://stackoverflow.com/questions/75092397/pythonanywhere-error-running-wsgi-application-declarativebase)

Thanks heaps for your time,

**Comment by zzzeek** [comment, marked as answer]
your live environment likely has a different version of SQLAlchemy running:

 ImportError: cannot import name 'DeclarativeBase' from 'sqlalchemy.orm' (/home/ziscoplankton/paychckr/venv/lib/python3.10/site-packages/sqlalchemy/orm/__init__.py)

take a look in that actual file and see what symbols are in there, it's likely a 1.4 version.   SQLAlchemy 2.0 is only installed when using `pip install` with the `--pre` option.

**Comment by ziscoplankton** [comment]
Hi @zzzeek,


Thank you for your time.

This solved the issue ! 

Therefore, I then ran into another one about the create_engine relative path with the three slashes but could fix it quickly. 

Then, I also had to change the session maker as sqllite only allows one session at all times. 

I created the main session with scoped_session into my db.py and imported it into the others.

It was a great experience doing it.

Thank you again for your time to help me resolve this issue.

May I ask how did you get to the conclusion of versions with the error line:
```
ImportError: cannot import name 'DeclarativeBase' from 'sqlalchemy.orm' (/home/ziscoplankton/paychckr/venv/lib/python3.10/site-packages/sqlalchemy/orm/init.py)
```

**Reply by CaselIT** [reply]
`DeclarativeBase` is only on v2, not in 1.4 so the import error only on some case suggested a version mismatch
